hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/common/AbstractStreamWriteFunction.java

```diff
@@ -265,6 +265,6 @@ protected String instantToWrite(boolean hasData) {
    * Returns whether the pending instant is invalid to write with.
    */
   private boolean invalidInstant(String instant, boolean hasData) {
-    return instant.equals(this.currentInstant) && hasData && !this.ckpMetadata.isAborted(instant);
+    return instant.equals(this.currentInstant) && hasData && this.ckpMetadata.isAborted(instant);
   }
```

Comment:
Seems not right ? do you mean `|| this.ckpMetadata.isAborted(instant);`

Reply:
Currently only when the checkpoint is aborted, we mark the instant as aborted to reuse the instant time.

Reply:
Yes it was not right .The aborted instant maybe also deleted by rollback if some write task failed , and other like checkpoint timeout we should reuse this instant . do you have some good idea

Reply:
Maybe we should delete the inflight instant when all write tasks failed